django/db/models/fields/related.py

```diff
@@ -463,9 +463,31 @@ def __init__(self, to, on_delete, from_fields, to_fields, rel=None, related_name
 
     def check(self, **kwargs):
         errors = super(ForeignObject, self).check(**kwargs)
+        errors.extend(self._check_to_fields_exist())
         errors.extend(self._check_unique_target())
         return errors
 
+    def _check_to_fields_exist(self):
+        if isinstance(self.remote_field.model, six.string_types):
+            return []
+
+        errors = []
+        for to_field in self.to_fields:
+            if to_field:
+                try:
+                    self.remote_field.model._meta.get_field(to_field)
+                except exceptions.FieldDoesNotExist:
+                    errors.append(
+                        checks.Error(
+                            "The related model '%s' has no field named '%s'."
```

Comment:
Would this message make it more clear about what argument of the field is problematic?
```
"The 'to_field' '%s' doesn't exist on the related "
"model '%s'."
% (to_field, self.remote_field.model._meta.label),
```